src/backends/beamcoder.ts

```diff
@@ -246,15 +246,22 @@ export class BeamcoderExtractor extends BaseExtractor implements Extractor {
      * Get imageData for a given time in seconds
      * @param targetTime
      */
-    async getImageDataAtTime(targetTime: number): Promise<ImageData> {
+    async getImageDataAtTime(targetTime: number, target?: Uint8ClampedArray): Promise<ImageData> {
         const targetPts = Math.round(this._timeToPTS(targetTime));
         VERBOSE && console.log('targetTime', targetTime, '-> targetPts', targetPts);
         const frame = await this._getFrameAtPts(targetPts);
         if (!frame) {
             VERBOSE && console.log('no frame found');
             return null;
         }
-        const rawData = this._resizeFrameData(frame);
+
+        let rawData = target;
+
+        if (!target) {
+            rawData = new Uint8ClampedArray(frame.width * frame.height * 4);
```

Comment:
We should probably unhardcode the `4` here too. It would make the code more self-explanatory.

Reply:
(`const components = /* whatever is needed to extract component number from the frame */;`)

Reply:
@antoineMoPa , I create constant for that
RGBA_PIXEL_SIZE